include/nbl/builtin/hlsl/tgmath/impl.hlsl

```diff
@@ -586,16 +536,57 @@ template<typename T>
 NBL_PARTIAL_REQ_TOP(concepts::FloatingPointScalar<T>)
 struct l2gamma_helper<T NBL_PARTIAL_REQ_BOT(concepts::FloatingPointScalar<T>) >
 {
-	// modified Stirling's approximation for log2 up to 3rd polynomial
+	// implementation derived from Numerical Recipes in C, transformed for log2
 	static T __call(T x)
 	{
-		const T thresholds[3] = {5e4, 1e19, 1e19};
-		if (x > thresholds[findLSB(uint32_t(sizeof(T)))])
+		// TODO: need a way to silence warning about thresholds being too large for T
```

Comment:
do the pragmas or compile options I suggested work? if yes then remove the comment